Fix wrong time format

tklecka opened the pull request:
closes #97
https://github.com/home-assistant/core/issues/83988

as @steinerl said 
it looks like the ZAMG changed their timestamp format.
I'm also not sure if this is a permanent change...

killer0071234 commented:
Please change also the test data in [data_station.json](https://github.com/killer0071234/python-zamg/blob/fa6809461932964b11852a49460b916814b0728f/tests/data_station.json?short_path=be9b80e#L5)

tklecka commented:
> Please change also the test data in [data_station.json](https://github.com/killer0071234/python-zamg/blob/fa6809461932964b11852a49460b916814b0728f/tests/data_station.json?short_path=be9b80e#L5)

Sorry didn't see that, WIP

steinerl commented:
The [metadata](https://dataset.api.hub.zamg.ac.at/v1/station/current/tawes-v1-10min/metadata) timestamp format has also changed, which means that [data_metadata.json](https://github.com/killer0071234/python-zamg/blob/a1c209e048740c9e67ce7baf9705721473d7c5ac/tests/data_metadata.json) is basically also incorrect.

tklecka commented:
> The [metadata](https://dataset.api.hub.zamg.ac.at/v1/station/current/tawes-v1-10min/metadata) timestamp format has also changed, which means that [data_metadata.json](https://github.com/killer0071234/python-zamg/blob/a1c209e048740c9e67ce7baf9705721473d7c5ac/tests/data_metadata.json) is basically also incorrect.

True, but I don't think the data from the metadata is used anywhere. I have changed them in the [data_metadata.json](https://github.com/killer0071234/python-zamg/blob/a1c209e048740c9e67ce7baf9705721473d7c5ac/tests/data_metadata.json), just in case.

steinerl commented:
@killer0071234 also curious why you don't parse the timezone offset like so `%Y-%m-%dT%H:%M%z` and instead use the hardcoded `+00:00` and set the time zone to `Europe/Vienna` afterwards, which means that the timestamps are incorrect (off by the Viennese time zone offset).
Any particular reason?

```python
>>> fixed_tz = datetime.strptime("2022-12-15T10:30+00:00", "%Y-%m-%dT%H:%M+00:00").replace(tzinfo=zoneinfo.ZoneInfo("Europe/Vienna")
datetime.datetime(2022, 12, 15, 10, 30, tzinfo=zoneinfo.ZoneInfo(key='Europe/Vienna'))

>>> parsed_tz = datetime.strptime("2022-12-15T10:30+00:00", "%Y-%m-%dT%H:%M%z")
datetime.datetime(2022, 12, 15, 10, 30, tzinfo=datetime.timezone.utc)

>>> parsed_tz - fixed_tz
datetime.timedelta(seconds=3600)
```

killer0071234 commented:
> @killer0071234 also curious why you don't parse the timezone offset like so `%Y-%m-%dT%H:%M%z` and instead use the hardcoded `+00:00` and set the time zone to `Europe/Vienna` afterwards, which means that the timestamps are incorrect (off by the Viennese time zone offset). Any particular reason?
> 
> ```python
> >>> fixed_tz = datetime.strptime("2022-12-15T10:30+00:00", "%Y-%m-%dT%H:%M+00:00").replace(tzinfo=zoneinfo.ZoneInfo("Europe/Vienna"))
> datetime.datetime(2022, 12, 15, 10, 30, tzinfo=zoneinfo.ZoneInfo(key='Europe/Vienna'))
> 
> >>> parsed_tz = datetime.strptime("2022-12-15T10:30+00:00", "%Y-%m-%dT%H:%M%z")
> datetime.datetime(2022, 12, 15, 10, 30, tzinfo=datetime.timezone.utc)
> 
> >>> parsed_tz - fixed_tz
> datetime.timedelta(seconds=3600)
> ```

This part of the code I took over from the old Zamg integration, and did not dig into it that deep

killer0071234 commented:
> > The [metadata](https://dataset.api.hub.zamg.ac.at/v1/station/current/tawes-v1-10min/metadata) timestamp format has also changed, which means that [data_metadata.json](https://github.com/killer0071234/python-zamg/blob/a1c209e048740c9e67ce7baf9705721473d7c5ac/tests/data_metadata.json) is basically also incorrect.
> 
> True, but I don't think the dates from the metadata are used anywhere. I have changed them in the [data_metadata.json](https://github.com/killer0071234/python-zamg/blob/a1c209e048740c9e67ce7baf9705721473d7c5ac/tests/data_metadata.json), just in case.

That's true, this part is just copied in from a real reply and we are just using the possible stations out of it.

killer0071234 commented on src/zamg/zamg.py:
```diff
@@ -178,9 +178,9 @@ def set_default_station(self, station_id: str):
-            return datetime.strptime(
-                self._timestamp, "%Y-%m-%dT%H:%M:%S+00:00"
-            ).replace(tzinfo=zoneinfo.ZoneInfo("Europe/Vienna"))
+            return datetime.strptime(self._timestamp, "%Y-%m-%dT%H:%M+00:00").replace(
+                tzinfo=zoneinfo.ZoneInfo("Europe/Vienna")
+            )
```
```suggestion
            return datetime.strptime(self._timestamp, "%Y-%m-%dT%H:%Mz").astimezone(
                tz=zoneinfo.ZoneInfo("Europe/Vienna")
            )
```

@steinerl Did you mean a change like this?
Then we are getting the `last_update` correct for Austria.

steinerl replied:
I don't see why we would need the `Europe/Vienna` timezone. The ZAMG timestamps are in `UTC` and as long as we create a datetime that includes the `UTC` timezone it should be correct.

That should be enough:
```python
return datetime.strptime(self._timestamp, "%Y-%m-%dT%H:%M%z")
```

killer0071234 replied:
Yes I agree, then we should take the solution from @steinerl.
@tklecka Please change it accordingly.
And also change line 192:
`> datetime.utcnow()`
Thanks

steinerl replied:
I would keep line 192 as is, because if you remove the `.replace(tzinfo=zoneinfo.ZoneInfo("UTC"))` part, you remove the timezone info from the datetime object.
